From 7d923bb0db3b9b7377cdc505307dd7a839be77bb Mon Sep 17 00:00:00 2001
From: Colin Watson <cjwatson@debian.org>
Date: Wed, 2 Apr 2014 16:47:48 +0100
Subject: Remove libpci dependency

<pci/pci.h> was being included, but not actually used in any meaningful
way.  Since libpci is distributed under the GPL which is not compatible
with iprutils' licence, it would be a good idea to avoid this
dependency.

iprutils was however relying on the u8, u16, u32, and u64 types defined
by <pci/types.h>.  Define these locally (using <stdint.h>) instead.

Forwarded: yes
Last-Update: 2014-04-02

Patch-Name: remove-libpci-dependency.patch
---
 iprlib.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/iprlib.h b/iprlib.h
index 50f07e5..4ae2fd4 100644
--- a/iprlib.h
+++ b/iprlib.h
@@ -18,6 +18,7 @@
 #include <stdarg.h>
 #include <stddef.h>
 #include <stdio.h>
+#include <stdint.h>
 #include <fcntl.h>
 #include <errno.h>
 #include <sys/mount.h>
@@ -40,7 +41,6 @@
 #include <ctype.h>
 #include <syslog.h>
 #include <term.h>
-#include <pci/pci.h>
 #include <stdbool.h>
 #include <netinet/in.h>
 #include <asm/byteorder.h>
@@ -50,6 +50,11 @@
 #include <linux/netlink.h>
 #include <time.h>
 
+typedef uint8_t u8;
+typedef uint16_t u16;
+typedef uint32_t u32;
+typedef uint64_t u64;
+
 #define IPR_DASD_UCODE_USRLIB                0
 #define IPR_DASD_UCODE_ETC                   1
 
